### PR TITLE
Solve ticket 07 - Post comment

### PR DESCRIPTION
**Pull request by alfonzodev**

### CODE REVIEW

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
Hey Jose, I've requested some changes here and some thoughts from you. Let me know what you think of what I've asked. 

Submit another PR when you're happy with your changes

File: __tests__/app.test.js [hunk 1]
```diff
@@ -157,6 +157,136 @@ describe("/api/articles/:article_id/comments", () => {
...
+            body: testComment.body,
+            article_id: 8,
+            author: testComment.username,
+            votes: 0,
+            created_at: expect.any(String),
+          });
+        });
+    });
+    test("400: responds with Not Found when provided a non-existing article_id", () => {
+      const testComment = {
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js:206]: this test you've written is actually testing for a 404 - ID is valid type but non-existing

File: __tests__/app.test.js [hunk 2]
```diff
@@ -157,6 +157,136 @@ describe("/api/articles/:article_id/comments", () => {
...
+        username: "lurker",
+        body: "This is the best article I've read thus far.",
+      };
+      return request(app)
+        .post("/api/articles/300/comments")
+        .send(testComment)
+        .expect(400)
+        .then(({ body }) => {
+          const { msg } = body;
+          expect(msg).toBe("Invalid article id!");
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js:216]: I'd maybe consider having a different error message for the 404 error, this ID is no invalid, it just doesn't exist in the database... yet

Comment (Pipe-Smoking-Rabbit) [reply __tests__/app.test.js:216]: These messages are for yourself, when you get to make a website in frontend you'll be using this API, if something goes wrong these error messages are designed to help you figure out why

File: __tests__/app.test.js (hunk 1 above)
Comment (alfonzodev) [reply __tests__/app.test.js:206]: Missed a few details in the refactor process 😅

File: __tests__/app.test.js [hunk 3]
```diff
@@ -157,6 +157,136 @@ describe("/api/articles/:article_id/comments", () => {
...
+        .post("/api/articles/not-a-number/comments")
+        .send(testComment)
+        .expect(400)
+        .then(({ body }) => {
+          const { msg } = body;
+          expect(msg).toBe("Invalid article id!");
+        });
+    });
+    test("400: responds with Not Found when username in request body does not exist", () => {
+      const testComment = {
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js:234]: This test is also a 404 strictly speaking, any time an error relates to information that cannot be found int he database, think `404`

**Review by Pipe-Smoking-Rabbit — APPROVED** [review]
Really nice coding of the controller and model! The tests are very thorough too! 👍 
I've left some feedback on a couple of test description relating to error codes to look over, review at your leisure then merge when ready

**Review by Pipe-Smoking-Rabbit — APPROVED** [review]
Smashed it! 🥇

**Pull request merged**